src/test/spec/initial_dns_seedlist_discovery.rs

```diff
@@ -68,44 +80,66 @@ async fn run() {
-        // spec JSON.
-        let resolved_db = test_file
-            .options
-            .as_ref()
-            .and_then(|opts| opts.auth_source.clone())
-            .or_else(|| {
-                test_file
-                    .parsed_options
-                    .as_ref()
-                    .and_then(|opts| opts.db.clone())
-            });
+        // "txt-record-with-overridden-ssl-option.json" requires SSL be disabled; see DRIVERS-1324.
+        let requires_tls = match test_file.options {
+            Some(ref options) => options.ssl,
+            None => true,
+        };
+        let client = TestClient::new().await;
+        if requires_tls == client.options.tls_options().is_some()
+            && client.is_replica_set()
+            && client.options.repl_set_name.as_deref() == Some("repl0")
+        {
+            // If the connection URI provides authentication information, manually create the user
+            // before connecting.
+            if let Some(ParsedOptions {
+                user: Some(ref user),
+                password: Some(ref pwd),
+                ref db,
+            }) = test_file.parsed_options
+            {
+                client
+                    .drop_and_create_user(
+                        user,
+                        pwd,
+                        &[],
+                        &[AuthMechanism::ScramSha1, AuthMechanism::ScramSha256],
+                        db.as_deref(),
+                    )
+                    .await
+                    .unwrap();
+            }
+
+            let mut options_with_tls = options.clone();
+            if requires_tls {
+                let tls_options = TlsOptions::builder()
+                    .allow_invalid_certificates(true)
+                    .build();
+                options_with_tls.tls = Some(Tls::Enabled(tls_options));
+            }
+            let mut client = TestClient::with_options(Some(options_with_tls)).await;
+            let mut actual_hosts = client.server_info.hosts.unwrap();
+            if let Some(ref mut arbiters) = client.server_info.arbiters {
+                actual_hosts.append(arbiters);
+            }
+
+            test_file.hosts.sort();
+            actual_hosts.sort();
+
+            assert_eq!(test_file.hosts, actual_hosts);
+        }
 
         if let Some(ref mut resolved_options) = test_file.options {
-            resolved_options.auth_source = resolved_db.clone();
-
-            let actual_options = ResolvedOptions {
-                replica_set: options.repl_set_name.clone(),
-                auth_source: options
-                    .credential
-                    .as_ref()
-                    .and_then(|cred| cred.source.clone()),
-                ssl: options.tls_options().is_some(),
-            };
-
-            assert_eq!(&actual_options, resolved_options);
+            // When an `authSource` is provided without any other authentication information, we do
```

Comment:
I filed DRIVERS-796 a while back to improve this and it was subsequently closed, but clearly it's still a problem for us. I added a comment to it to try to re-open it so we can properly assert here.